transport-native-kqueue/src/main/java/io/netty/channel/kqueue/KQueueDatagramChannel.java

```diff
@@ -399,48 +384,31 @@ public KQueueDatagramChannelConfig config() {
 
     @Override
     protected void doDisconnect() throws Exception {
+        InetSocketAddress address = remoteAddress();
+        socket.disconnect();
         connected = false;
+        if (address != null) {
```

Comment:
same comment as https://github.com/netty/netty/pull/6193#discussion_r124030018

> @normanmaurer - https://github.com/netty/netty/pull/6193#discussion_r115565539 has been addressed ... but I'm still wondering if we should unconditionally set active to false here instead of depending on `remoteAddress()`?